syntax = "proto3";
package replication;

// A replication update from one replica to another
message ReplicationUpdateRequest {
  // A unique update identifier to ensure idempotency
  string update_id = 1;
  // Type of update (e.g., "create_account", "send_message", etc.)
  string update_type = 2;
  // JSON-encoded data for the update
  string data = 3;
}

message ReplicationUpdateResponse {
  bool success = 1;
  string message = 2;
}

service ReplicationService {
  rpc ReplicateUpdate (ReplicationUpdateRequest) returns (ReplicationUpdateResponse);
}
